eclipse-theia/theia
# rework monaco commands to align with VS Code

#### What it does


- Reverts https://github.com/eclipse-theia/theia/commit/40b1d45d45fdfc6b1db47c0690e450c5b28eb467 to align with VS Code behaviour as before
- Refactors monaco-commands to fix Find/Replace/Undo/Redo/Select All, but at the same time without breaking previous behaviour: https://github.com/eclipse-theia/theia/blob/42ea5bd196d6efa579ed571e64f7fc2e5e269e19/packages/monaco/src/browser/monaco-command.ts#L80-L187
- Adds integration tests for Find/Replace/Undo/Redo/Select All commands

#### How to test


- Find/Replace -> if the current editor then it should be used, if not then either suppressed or a widget should be used, i.e. the terminal
  - integration tests added to cover common cases
- Undo/Redo/Select All -> if the current editor then it should be used, otherwise delegate to DOM, rely on Monaco for implementation. Check input areas with and without editors in other views, git, SIW, terminal find, webview (gitlens welcome)
  - integration tests added to cover common cases
- Editor Actions like manipulation of selections
  - should be available in the command palette when the current editor is available and work properly
  - language features should be disable when they are not provided even for the current editor
  - test inline editors as debug console input and breakpoint expression editor
  - test embedded editors as peek references
- there should NOT be any exceptions in logs about missing service if the gitlens extension is installed
- integration tests should pass

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
